src/Controls/src/Core/ShellToolbar.cs

```diff
@@ -166,7 +166,7 @@ void OnBackButtonPropertyChanged(object sender, System.ComponentModel.PropertyCh
 			void OnBackButtonCanExecuteChanged(object sender, EventArgs e)
```

Comment:
I think the issue might be related to these being local functions. Can you move all of these so they are class instance methods opposed to local functions?